portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +26,18 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> test = new ArrayList<String>();
```

Comment:
Prefer Interface type to implementation type on the variable definition.

Also, this is not a testing class, maybe name it as "demoGreetings" or "hardcodedGreetings".

Reply:
Variable name made more clear. Also, why Interface type preferred?

Reply:
The high level idea is that you should use a variable by its definition instead of implementation. By doing that, it gives you flexibility to change the implementation in future without affecting the rest of the code.

e.g. if you initially define a TreeMap, you may assume it is a tree and its key is ordered. Later you may switch to a different data source that uses HashMap. Now you have to change all codes that based on that assumption(because the key now is not unsorted and you won't get method like `firstKey()`). If you stick with the interface Map and only use common methods like get and put. then changing implementation won't be a problem.

Reply:
Ah ok I see, the flexibility is definitely preferred. Thank you!